include/libtorrent/alert_types.hpp

```diff
@@ -2507,12 +2507,30 @@ namespace libtorrent
 #endif // TORRENT_DISABLE_LOGGING
 	};
 
+	// This alert is generated when a piece has passed the hash check and has
+	// been written to disk (so it can be safely read). This alert may be generated
+	// more than once for each piece.
```

Comment:
why may it be generated more than once per piece? is there anything we can do to avoid that?

Reply:
I don't think it will now that I'm getting a better picture of how libtorrent works so I'll change that.